### PR TITLE
Locale LanguageFilter translation implemented

### LINKED ISSUE #247: Improve Course Language Filter
Let's add the language name in the [device's primary locale](https://stackoverflow.com/a/39639103/4870048) as a subtitle to the course language filter as iOS is doing it. We're currently only displaying the language name in its native locale.

Also, please cleanup `CourseFilterView` and `CourseFilterClassifierView`, e.g., the workaround for wrong Chinese API codes should be handled by a new method in `LanguageUtil`. Probably, also a custom layout is required for this filter.

Screen Shot 2020-05-14 at 17 09 06

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #247 

Changes added to support a translation of course languages to the locale language. The format is nativeLanguage (localLanguage) f.e. Espagnol (Spanisch). Additionally the `LanguageUtil` was cleaned up.

### CODE REVIEW

File: app/src/main/java/de/xikolo/utils/LanguageUtil.kt [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
     }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    fun toLocaleName(context: Context, tag: String): String {
+        val languageTag = correctLanguageTag(tag)
+        val locale = Locale.forLanguageTag(languageTag)
+
+        val deviceLocale =
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                context.resources.configuration.locales.get(0)
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/utils/LanguageUtil.kt:25]: You are using the `context` here to fetch the device language, yet in `deviceLanguage` you are using `Locale.getDefault()`. I think this should be consistent.

**Review by bengelhaupt — CHANGES_REQUESTED** [review]
Simple, yet effective I think. Good work!

File: app/src/main/java/de/xikolo/utils/LanguageUtil.kt (hunk 1 above)
Comment (SanJSp) [reply app/src/main/java/de/xikolo/utils/LanguageUtil.kt]: Found a solution to use even less code :) maybe you can have a look

File: app/src/main/java/de/xikolo/utils/LanguageUtil.kt [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
 
         val locale = Locale.forLanguageTag(languageTag)
         return locale.getDisplayLanguage(locale).capitalize(locale)
     }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    fun toLocaleName(context: Context, tag: String): String {
+        val languageTag = correctLanguageTag(tag)
+        val locale = Locale.forLanguageTag(languageTag)
+        val deviceLocale = Locale(Locale.getDefault().language)
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/utils/LanguageUtil.kt:21]: Is this necessary, wouldn't `Locale.getDefault()` suffice?

**Review by bengelhaupt — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/xikolo/utils/LanguageUtil.kt (hunk 2 above)
Comment (SanJSp) [reply app/src/main/java/de/xikolo/utils/LanguageUtil.kt:21]: Yes, it suffices :) good catch!

**Review by tobiasrohloff — APPROVED** [review]

**Pull request merged**